source/core/Events.cpp

```diff
@@ -396,6 +409,42 @@ class EventRegistrationRefNumManager : public RefNumManager {
 
 EventRegistrationRefNumManager EventRegistrationRefNumManager::_s_singleton;
 
+// GetPendingEventInfo -- given a static queue and a set of dynamic event queues (passed via refnum), return the queue with the earliest event
+// based on sequence number.  Returns 0 if the earliest event is in the static queue (if non-null), or a 1-based index into the dynamic queue
+// list if the earliest event is in one of the dynamic queues.
+// Also returns the base dynamic index of the returned queueID (e.g. event reg. refnum; if this event reg. refnum has multiple registered items
+// the dynIndex will be further incremented by the caller to indicate which item actually matches the event).
+Int32 EventOracle::GetPendingEventInfo(EventQueueID *pActiveQID, Int32 nQueues, RefNumVal *dynRegRefs, Int32 *dynIndexBase) {
+    EventQueueID eventQID = pActiveQID ? *pActiveQID : kNotAQueueID;
+    Int32 earliestIndex = -1, regIndex = 0;
+    UInt32 earliestSeq = 0, eventSeq = 0;
+    Int32 skippedEntries = 0;
+    if (eventQID) {  // passed static QueueID to check with dynamic set
+        if (_qObject[eventQID].GetPendingEventSequenceNumber(&eventSeq)) {
+            earliestIndex = 0;
+            earliestSeq = eventSeq;
+        } else {
+            eventQID = kNotAQueueID;
+        }
+    }
+    while (regIndex < nQueues) {
+        DynamicEventRegInfo *regInfo = NULL;
+        if (dynRegRefs
+            && EventRegistrationRefNumManager::RefNumStorage().GetRefNumData(dynRegRefs[regIndex].GetRefNum(), &regInfo) == kNIError_Success) {
+            if (_qObject[regInfo->_qID].GetPendingEventSequenceNumber(&eventSeq) && (!eventQID || EventTimeSeqCompare(eventSeq, earliestSeq) < 0)) {
```

Comment:
For long if statements consider using a local boolean named with the intent of the check, basically the name should answer the question, what is the code checking here? and then use the boolean in the if statement. This is helpful for readability as long as it doesn't have a performance impact.

Apply this rule where you consider is useful in other sections of the code.

Reply:
I agree in general, but this is used in dispatching events, so I am actually concerned about performance.  I have added more comments to help readability, though.